pkg/driver/aws-efs/aws_efs.go

```diff
@@ -0,0 +1,120 @@
+					Name:                "driver-m",
+				},
+			},
+			SidecarLocalMetricsPortStart:   commongenerator.AWSEFSLoopbackMetricsPortStart + 1,
+			SidecarExposedMetricsPortStart: commongenerator.AWSEFSExposedMetricsPortStart + 1,
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultProvisioner.WithExtraArguments(
+					"--feature-gates=Topology=true",
+					"--extra-create-metadata=true",
+					"--timeout=5m",
+					"--worker-threads=1",
+				),
+				commongenerator.DefaultLivenessProbe.WithExtraArguments(
+					"--probe-timeout=3s",
+				),
+			},
+			Assets: commongenerator.DefaultControllerAssets.WithAssets(generator.StandaloneOnly,
+				"overlays/aws-efs/base/privileged_role.yaml",
+				"overlays/aws-efs/base/controller_privileged_binding.yaml",
+				"overlays/aws-efs/base/credentials.yaml",
+			),
+		},
+
+		GuestConfig: &generator.GuestConfig{
+			DaemonSetTemplateAssetName:   "overlays/aws-efs/patches/node_add_driver.yaml",
+			LivenessProbePort:            10303,
+			NodeRegistrarHealthCheckPort: 10305,
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultNodeDriverRegistrar,
+				commongenerator.DefaultLivenessProbe.WithExtraArguments(
+					"--probe-timeout=3s",
+				),
+			},
+			Assets: commongenerator.DefaultNodeAssets.WithAssets(generator.StandaloneOnly,
+				"overlays/aws-efs/base/csidriver.yaml",
+			),
+			AssetPatches: generator.NewAssetPatches(generator.StandaloneOnly,
+				// Any role or cluster role bindings should not hardcode service account namespace because this operator is OLM based and can be installed into a custom namespace.
+				"main_provisioner_binding.yaml", "overlays/aws-efs/patches/binding_with_namespace_placeholder.yaml",
+				"lease_leader_election_binding.yaml", "overlays/aws-efs/patches/binding_with_namespace_placeholder.yaml",
+			),
+		},
+	}
+}
+
+// GetAWSEFSOperatorConfig returns runtime configuration of the CSI driver operator.
+func GetAWSEFSOperatorConfig() *config.OperatorConfig {
+	return &config.OperatorConfig{
+		CSIDriverName:                   opv1.AWSEFSCSIDriver,
+		UserAgent:                       "aws-efs-csi-driver-operator",
+		AssetReader:                     assets.ReadFile,
+		AssetDir:                        generatedAssetBase,
+		OperatorControllerConfigBuilder: GetAWSEFSOperatorControllerConfig,
+		Removable:                       false,
+	}
+}
+
+// GetAWSEFSOperatorControllerConfig returns second half of runtime configuration of the CSI driver operator,
+// after a client connection + cluster flavour are established.
+func GetAWSEFSOperatorControllerConfig(ctx context.Context, flavour generator.ClusterFlavour, c *clients.Clients) (*config.OperatorControllerConfig, error) {
+	cfg := operator.NewDefaultOperatorControllerConfig(flavour, c, "AWSEFS")
+	cfg.AddDeploymentHookBuilders(c, withCABundleDeploymentHook)
+	cfg.DeploymentWatchedSecretNames = append(cfg.DeploymentWatchedSecretNames, cloudCredSecretName, metricsCertSecretName)
```

Comment:
@RomanBednar I guess seems we miss the proxy hook to inject the cluster proxy to the controller and driver node, right?
https://github.com/openshift/aws-efs-csi-driver-operator/blob/master/pkg/operator/starter.go#L118

Reply:
I don't think we need the hook for node (daemon set) since it was not used in legacy code of EFS operator. And for controller (deployment) the hook is set by default for standalone flavour: https://github.com/openshift/csi-operator/blob/95ade5e482603c8090cb5f39e69ca64db942ccbe/pkg/driver/common/operator/hooks.go#L37

Reply:
Thanks for the clarify! I checked the proxy config CI job looks good the proxy inject to the driver controller while the driver node does not need the proxy inject also works.